### PR TITLE
chore: Remove the subsumption inference used for label polymorphism

### PR DESCRIPTION
**Pull request by Marwes**
Removes the old subsumption (subtyping) based label polymorphism in preparation of https://github.com/influxdata/flux/issues/4927, https://github.com/influxdata/flux/issues/4928 and https://github.com/influxdata/flux/issues/4929 . Tests that fail are ignored or removed if they assumed subtyping.

(Re-creation of https://github.com/influxdata/flux/pull/4776 to get a clean slate since that PR has changed so much)

### Checklist

Dear Author :wave:, the following checks should be completed (or explicitly dismissed) before merging.

- [x] ✏️ Write a PR description, regardless of triviality, to include the _value_ of this PR
- [x] 🔗 Reference related issues
- [x] 🏃 Test cases are included to exercise the new code
- [x] 🧪 If **new packages** are being introduced to stdlib, link to Working Group discussion notes and ensure it lands under `experimental/`
- [x] 📖 If **language features** are changing, ensure `docs/Spec.md` has been updated

Dear Reviewer(s) :wave:, you are responsible (among others) for ensuring the completeness and quality of the above before approval.

### CODE REVIEW

File: interpreter/interpreter_test.go [hunk 1]
```diff
@@ -883,8 +883,8 @@ func TestStack(t *testing.T) {
 			FunctionName: "window",
 			Location: ast.SourceLocation{
 				File:   "universe/universe.flux",
-				Start:  ast.Position{Line: 3814, Column: 12},
-				End:    ast.Position{Line: 3814, Column: 51},
+				Start:  ast.Position{Line: 4031, Column: 12},
+				End:    ast.Position{Line: 4031, Column: 51},
```
Comment (wolffcm) [inline interpreter/interpreter_test.go:887]: I don't see any changes to the stdlib other than removing the `experimental/universe` package, so why would this test need to change?

File: stdlib/experimental/universe/universe.flux [hunk 2]
```diff
@@ -1,122 +0,0 @@
...
-//
-// ## Metadata
-// introduced: v0.166.0
-// tags: transformations, aggregates
-//
-builtin mean : (<-tables: stream[{A with C: B}], ?column: C = "_value") => stream[{C: B}]
-    where
-    A: Record,
-    B: Numeric,
-    C: Label
```
Comment (wolffcm) [inline stdlib/experimental/universe/universe.flux:122]: In the header comment for the PR you say
> I also added the updated type signatures which are hidden by `@feature` usage so they do not appear in the current standard library.

But I don't see any changes like that here; only the deleted package `experimental/universe`. Am I missing something?

**Review by wolffcm — COMMENTED** [review]
This looks good to me. Based on your PR comment I would have expected existing stdlib functions to be annotated with `@feature` so either you didn't actually do this or I'm misunderstanding something.

File: stdlib/experimental/universe/universe.flux (hunk 2 above)
Comment (Marwes) [reply stdlib/experimental/universe/universe.flux:122]: It was actually split out to https://github.com/influxdata/flux/pull/4976 (I forgot to remove that section in the PR message)

File: interpreter/interpreter_test.go (hunk 1 above)
Comment (Marwes) [reply interpreter/interpreter_test.go]: Forgot to revert the line after splitting out https://github.com/influxdata/flux/pull/4976 . Fixed.

**Review by wolffcm — APPROVED** [review]

**Review by sunbryely-influxdata — APPROVED** [review]

**Pull request merged**